src/main/java/org/apache/maven/shared/transfer/dependencies/resolve/internal/Maven30DependencyResolver.java

```diff
@@ -153,12 +154,12 @@
         final Class<?>[] argClasses =
             new Class<?>[] { org.apache.maven.model.Dependency.class, ArtifactTypeRegistry.class };
 
-         List<Dependency> aetherDependencies = new ArrayList<Dependency>( mavenDependencies.size() );
+         List<Dependency> aetherDependencies = mavenDependencies != null
+                                                 ? new ArrayList<Dependency>( mavenDependencies.size() )

```

Comment:
This is way more complex than needed: just return a new ArrayList without a size. 

Do **not** return `Collections.emptyList`. That list is immutable.